src/apps/VirtualApp/VirtualChat.js

```diff
@@ -150,13 +172,36 @@ class VirtualChat extends Component {
   };
 
   sendChatMessage = () => {
-    let { input_value, user, from, room_chat, support_msgs } = this.state;
-    if (!user.role.match(/^(user|guest)$/) || input_value === '') {
+    const { input_value, user: {role, display, username} } = this.state;
+    if (!role.match(/^(user|guest)$/) || input_value === '') {
       return;
     }
-    let msg     = { user, text: input_value };
-    let pvt     = room_chat ? '' : from ? { 'to': from } : { 'to': `${SHIDUR_ID}` };
-    let message = {
+    const msg = { user: {role, display, username}, text: input_value };
+    const onError = (reason) => {
+      alert(reason);
+    };
+    const onSuccess = (msg) => {
+      const { room_chat, support_msgs } = this.state;
+      Janus.log(':: Message sent ::');
+      this.setState({ input_value: '' });
+      if (!room_chat) {
+        support_msgs.push(msg);
+        this.setState({ support_msgs });
+      }
+    }
+    this.state.gdm.sendGuaranteeMessage(msg, (m) => this.sendChatMessage_(m)).then(() => {
```

Comment:
`(m) => this.sendChatMessage_(m)` could be simply `this.sendChatMessage_`

Reply:
Removed this usage example, not really needed.